src/devices/Ak8963/Ak8963.cs

```diff
@@ -0,0 +1,275 @@
+        }
+
+        /// <summary>
+        /// Get the device information
+        /// </summary>
+        /// <returns></returns>
+        public byte GetDeviceInfo()
+        {
+            return ReadByte(Register.INFO);
+        }
+
+        /// <summary>
+        /// Get the magnetometer bias
+        /// </summary>
+        public Vector3 MagnometerBias
+        {
+            get { return _magnetometerBias; }
+            set { _magnetometerBias = value; }
+        }
+
+        /// <summary>
+        /// Calibrate the magnetometer. Make sure your sensor is as far as possible of magnet
+        /// Calculate as well the magnetometer bias
+        /// </summary>
+        /// <returns>Returns the factory calibration data</returns>
+        public Vector3 CalibrateMagnetometer()
+        {
+            Vector3 calib = new Vector3();
+            Span<byte> rawData = stackalloc byte[3];
+
+            var oldPower = MeasurementMode;
+
+            // Stop the magetometer
+            MeasurementMode = MeasurementMode.PowerDown;
+            // Enter the magnetometer Fuse mode to read the calibration data
+            MeasurementMode = MeasurementMode.FuseRomAccess;
+            // Read the data
+            ReadByteArray(Register.ASAX, rawData);
+            calib.X = (float)((rawData[0] - 128) / 256.0 + 1.0);
+            calib.Y = (float)((rawData[1] - 128) / 256.0 + 1.0);
+            calib.Z = (float)((rawData[2] - 128) / 256.0 + 1.0);
+            MeasurementMode = MeasurementMode.PowerDown;
+            MeasurementMode = oldPower;
+
+            // Now calculate the bias
+            // Store old mode to restore after
+            var oldMode = MeasurementMode;
+            Vector3 minbias = new Vector3();
+            Vector3 maxbias = new Vector3();
+            int numberMeasurements = 1500;
+
+            // Setup the 100Hz continous mode
+            MeasurementMode = MeasurementMode.ContinousMeasurement100Hz;
+            for (int reading = 0; reading < numberMeasurements; reading++)
+            {
+                var bias = ReadMagnetometer(true);
+                minbias.X = Math.Min(bias.X, minbias.X);
+                minbias.Y = Math.Min(bias.Y, minbias.Y);
+                minbias.Z = Math.Min(bias.Z, minbias.Z);
+                maxbias.X = Math.Max(bias.X, maxbias.X);
+                maxbias.Y = Math.Max(bias.Y, maxbias.Y);
+                maxbias.Z = Math.Max(bias.Z, maxbias.Z);
+                // 10 ms = 100Hz, so waiting to make sure we have new data
+                Thread.Sleep(10);
+            }
+            // Store the bias
+            _magnetometerBias = ((maxbias + minbias) / 2) + calib;
+
+            return calib;
+        }
+
+        /// <summary>
+        /// True if there is a data to read
+        /// </summary>
+        public bool WaitForDataReady => (ReadByte(Register.ST1) & 0x01) == 0x01;
+
+        /// <summary>
+        /// Check if the version is the correct one (0x48)
```

Comment:
Which version is this refering to? Do you mind adding more details in the summary as to when would you call this and why could this ever return false?